docs/CONTRIBUTING.md

```diff
@@ -279,6 +279,31 @@ python scripts/validate_metadata.py
 pre-commit run --all-files
 ```
 
+### Base Image Validation
+
+All components and pipelines must use approved base images. The validation script compiles components using `kfp.compiler` to extract the actual runtime images, which correctly handles:
+
+- Variable references (`base_image=MY_IMAGE`)
+- `functools.partial` wrappers
+- Default image resolution
+
+**Valid base images:**
+- Images starting with `ghcr.io/kubeflow/` (Kubeflow official registry)
+- Standard Python images (`python:<version>`, e.g., `python:3.11`, `python:3.11-slim`)
+
+Run the validation locally:
+
+```bash
+# Run with default settings
+uv run python .github/scripts/validate_base_images/validate_base_images.py
+
+# Override the allowed prefix (for testing or forks)
+uv run python .github/scripts/validate_base_images/validate_base_images.py \
+    --allowed-prefix ghcr.io/myorg/
```

Comment:
can move this outside of .github, we should only store files here if they are only meant for workflows, if we expect users to run them outside of a github workflow then `scripts/` might be more appropriate